core/src/main/scala/io/finch/endpoint/header.scala

```diff
@@ -1,39 +1,46 @@
 package io.finch.endpoint
 
+import scala.reflect.ClassTag
+
+import com.twitter.util.{Return, Throw}
 import io.finch._
 import io.finch.internal._
 import io.finch.items._
 
-private abstract class Header[A](name: String) extends Endpoint[A] {
+private abstract class Header[A](name: String)(implicit d: DecodeEntity[A], tag: ClassTag[A]) extends Endpoint[A] {
+  self =>
 
   protected def missing(input: Input, name: String): Endpoint.Result[A]
-  protected def present(input: Input, value: String): Endpoint.Result[A]
+  protected def present(input: Input, value: A): Endpoint.Result[A]
 
   final def apply(input: Input): Endpoint.Result[A] =
     input.request.headerMap.getOrNull(name) match {
       case null => missing(input, name)
-      case value => present(input, value)
+      case value => d(value) match {
+        case Return(r) => present(input, r)
+        case Throw(e) => notParsed(self, input, e, tag)
```

Comment:
I'd prefer we introduce `Rs.paramNotParsed` so we can stop relying on request items (I really really want them to go away in a long run; #787 should replace them). See #873 for example.